spec/consensus/light-client.md

```diff
@@ -316,3 +316,70 @@ func Backwards(h1,h2) bool {
   return (hash(h2) == old.Header.hash)
  }
 ```
+
+### Failure Detection Module
+
+The light client must be able to detect failures in the trust assumptions.
+Ideally, it should never trust a header, which was forged by malicious
+validators.
+
+### Connectivity requirement
+
+To meet the above requirement, it must be connected to at least one honest
+full node. This is sufficient so long as there's no fork on the main chain.
+
+If there is a fork on the main chain, it means that two full nodes have decided
+on two different headers for the same height. Then the requirement would at
+least require that the lite client is **connected to one honest full node on
+each branch of the fork**.
+
+_Assumption_: no more than `2f` of the nodes are malicious (`f+1` are honest).
+otherwise, they could create more forks.
+
```

Comment:
I don't think we discussed that. But my first guess is that Byzantine faults could collude to create one branch for each correct validator. Even if per consensus they can only branch into two: they could fork into two at height h. Then the two validator sets for height h+1 might be disjoint, except of the faulty nodes. They might create more forks on each branch, etc.

Reply:
> Even if per consensus they can only branch into two

they can only do that if they have `2f+1` (+2/3). otherwise, they can't change validator sets (`ValidatorsHash`), they can only vote for proposals from past rounds say (flip-flopping)

Reply:
Sorry. The above statement is incorrect. +1/3 can create as many forks as it wants (by not voting for 2/3 proposals and waiting say 10 rounds, and later constructing a fork for each round)

Reply:
I also discussed with Igor. We will use the model checker to see whether we can generate executions of consensus with several forks.